osf/external/gravy_valet/gv_mocks.py

```diff
@@ -0,0 +1,477 @@
+import contextlib
+import itertools
+import json
+import typing
+import re
+import urllib.parse
+
+import dataclasses  # backport
+import responses
+
+from . import auth_helpers
+from osf.models import OSFUser, AbstractNode
+from osf.utils import permissions as osf_permissions
+from website import settings
+
+
+@dataclasses.dataclass
+class _MockGVEntity:
```

Comment:
naming nitpick (feel free to disregard): could be confusing to call these things "mocks", since they have nothing to do with the standard `Mock` interface (with `assert_called` and friends) -- the word's not wrong, but the collision with a standard-lib term could be misleading...